support-frontend/app/controllers/CreateSubscription.scala

```diff
@@ -85,8 +85,15 @@ class CreateSubscription(
       deliveryAddress = request.deliveryAddress,
       telephoneNumber = request.telephoneNumber,
       allowMembershipMail = false,
-      allowThirdPartyMail = user.statusFields.flatMap(_.receive3rdPartyMarketing).getOrElse(false),
-      allowGURelatedMail = user.statusFields.flatMap(_.receiveGnmMarketing).getOrElse(false),
+      // Previously the values for the fields allowThirdPartyMail and allowGURelatedMail
+      // were derived by looking for the fields: statusFields.receive3rdPartyMarketing and
+      // statusFields.receiveGnmMarketing in the JSON object that models a user.
+      // However, a query of the identity database indicates that these fields aren't defined for any users
```

Comment:
nice work working all that out, so we were always writing false into Salesforce, regardless of people's actual preference?  And I think there is a job in membership-workflow that writes changes to these fields back into identitty and possibly vice versa.  I'm not sure if that job would be making things worse or better!  I'm not sure if those fields are used by the call centre in salesforce, and whether they affect braze!
Not sure if anyone else knows more about this off hand?

Reply:
Yeah, for `receiveGnmMarketing` I ran the following query against the PROD database:
```sql
select *
from users
where jdoc #>> '{statusFields,receiveGnmMarketing}' is not null
```
and it returned no results; same for `receive3rdPartyMarketing`. So I'd be surprised if it wasn't false for all cases. 

This is the one of the reasons why I wanted to remove the user models from identity-play-auth; its just another place where we have to keep the models up to date. The previously used [`StatusFields`](https://github.com/guardian/identity-auth/blob/1c6869141955b3da31675a3200c607f6ebeee2eb/src/main/scala/com/gu/identity/play/IdUser.scala#L5) were last updated 4 years ago.